### PR TITLE
[JBPM-9831] Use the encrypted values for the user and/or keystore password when starting smart router

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9831

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by gmunozfe** [comment]
jenkins retest this

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/handlers/AdminHttpHandler.java [hunk 1]
```diff
@@ -45,16 +46,10 @@
...
-
 public class AdminHttpHandler implements HttpHandler {
     
     private static final Logger log = Logger.getLogger(AdminHttpHandler.class);
 
-    private String CONTROLLER = System.getProperty(KIE_CONTROLLER);
-    private int interval = Integer.parseInt(System.getProperty(KIE_SERVER_CONTROLLER_ATTEMPT_INTERVAL, "5"));
-    private int attemptsLimit = Integer.parseInt(System.getProperty(KIE_SERVER_RECOVERY_ATTEMPT_LIMIT, "100"));
-    
 //    private KieServerProxyClient proxyClient;
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/handlers/AdminHttpHandler.java:53]: You could remove also this commented line

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/repository/ConfigFileWatcher.java [hunk 2]
```diff
@@ -29,13 +29,11 @@
 import org.jboss.logging.Logger;
 import org.kie.server.router.Configuration;
 import org.kie.server.router.KieServerRouter;
-import org.kie.server.router.KieServerRouterConstants;
+import org.kie.server.router.KieServerRouterEnvironment;
 
 public class ConfigFileWatcher implements Runnable {
     
     private static final Logger log = Logger.getLogger(KieServerRouter.class);
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/repository/ConfigFileWatcher.java:36]: Update this logger
```suggestion
    private static final Logger log = Logger.getLogger(ConfigFileWatcher.class);
```

**Review by gmunozfe — APPROVED** [review]
Looks good to me, fantastic work @elguardian !
Some minor comments, and also missing a test using the new config file, if you can add it.

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java [hunk 3]
```diff
@@ -193,26 +163,28 @@ public void start(String host,
               listeners);
     }
 
+    public void start(ConfigurationListener... listeners) {
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java:166]: The listeners should be passed in the method call below.

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java [hunk 4]
```diff
@@ -72,74 +57,61 @@
...
 
     public KieServerRouter() {
-        this(reloadManagementSecured(), reloadIdentityProvider()); //Reload for easier testing
+        this(new KieServerRouterEnvironment());
     }
 
-    public KieServerRouter(boolean isSecured, String identityServiceName) {
+    public KieServerRouter(KieServerRouterEnvironment env) {
         configRepositoryServiceLoader.forEach(repo -> repository = repo);
         log.info("KIE Server router repository implementation is " + repository);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java:104]: Should be placed after `this.repository = new FileRepository(env);`

**Review by sutaakar — APPROVED** [review]
Looks good to me once the comments are addressed.

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouterEnvironment.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                sb.append("http://");
+            } else {
+                sb.append("https://");
+            }
+            sb.append(routerHost);
+            sb.append(":");
+            if (httpEnabled) {
+                sb.append(port);
+            } else {
+                sb.append(sslPort);
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouterEnvironment.java:165]: https://github.com/kiegroup/droolsjbpm-integration/blob/ca1cbe5b28b55be11bce71c76478825493dcd07c/kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/handlers/KieServerInfoHandler.java#L125-L147

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java:166]: done

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/repository/ConfigFileWatcher.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/repository/ConfigFileWatcher.java:36]: done

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java (hunk 4 above)
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java:104]: done

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/handlers/AdminHttpHandler.java (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/handlers/AdminHttpHandler.java:53]: removed

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouterEnvironment.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouterEnvironment.java:165]: it is back compatible.

**Pull request merged**